activities.el

```diff
@@ -801,6 +819,90 @@ activity's name is NAME."
                   "In the meantime, it's recommended to not use buffers of this major mode in an activity's layout; or you may simply ignore this error and use the other buffers in the activity.")
           (current-buffer)))))
 
+(defvar activities--age-spec
+  `((?Y "year"   "years"   ,(round (* 60 60 24 365.2425)))
+    (?M "month"  "months"  ,(round (* 60 60 24 30.436875)))
+    (?w "week"   "weeks"   ,(* 60 60 24 7))
+    (?d "day"    "days"    ,(* 60 60 24))
+    (?h "hour"   "hours"   ,(* 60 60))
+    (?m "minute" "minutes" 60)
+    (?s "second" "seconds" 1))
+  "Age specification.  See `magit--age-spec', which this duplicates.")
+
+(defun activities--age (age &optional abbreviate)
+  "Summarize AGE and possibly ABBREVIATE.
+Adapted from `magit--age'."
+  (cl-labels ((fn (age spec)
+                (pcase-let ((`(,char ,unit ,units ,weight) (car spec)))
+                  (let ((cnt (round (/ age weight 1.0))))
+                    (if (or (not (cdr spec))
+                            (>= (/ age weight) 1))
+                        (list cnt (cond (abbreviate char)
+                                        ((= cnt 1) unit)
+                                        (t units)))
+                      (fn age (cdr spec)))))))
+    (fn age activities--age-spec)))
+
+(defun activities--oldest-age (activities)
+  "Return the age in seconds of the oldest activity in ACTIVITIES."
+  (cl-loop for (_name . act) in activities maximize
+	   (cl-loop
+	    for type in '(default last)
+	    for func = (intern (format "activities-activity-%s" type))
+	    maximize
+	    (float-time
+	     (time-since
+	      (map-elt (activities-activity-state-etc (funcall func act)) 'time))))))
+
+(defun activities-annotate (max-age oldest-possible)
+  "Return an activity annotation function.
+MAX-AGE is the maximum age of any activity in seconds.
+OLDEST-POSSIBLE is the oldest age in the `vc-annotate-color-map'."
+  (lambda (name)
+    (when-let ((activity (map-elt activities-activities name)))
+      (let (data (age-len 14))
+	(dolist (type '(last default))
+	  (let* ((func (intern (format "activities-activity-%s" type)))
+		 (state (funcall func activity))
+		 (time (map-elt (activities-activity-state-etc state) 'time))
+		 (window-state (activities-activity-state-window-state state))
+		 (buffers (window-state-buffers window-state))
+		 (files ( activities--map-window-state-leafs window-state
```

Comment:
It's actually not extraneous: it prevents emacs-lisp-mode from indenting the subsequent line way over to the right under `window-state`.   It's a reasonably new feature.  But it is ugly, so I hacked it out.  How do you deal with this problem (I notice you often keep the 1st argument on the line with the function call)?

Reply:
Yes, but it's intended for, e.g. quoted plists, not for function calls.  I guess one could use it if one preferred, but it would be unidiomatic and visually confusing, as it would tend to obscure that it's a function call.

I don't like having one-symbol lines, so I avoid them when possible.  It's a balancing act between that and line length.

It's generally fine to fix such places manually, but in my projects I use `makem.sh` with its `lint-indentation` rule, which will give a warning about manually adjusted indentation, so please just indent it as Emacs automatically does and we'll live with it.

Reply:
I do agree it's ugly, possibly uglier than one-symbol lines, but I tend to air on the compactness=readability side.  Restored the 1sl for now to be sure not to add linting errors.